src/commands/runDraftTool/runDraftDeployment.ts

```diff
@@ -128,27 +128,56 @@ async function multiStepInput(
     input: MultiStepInput,
     state: Partial<State>,
     step: number
-  ) {
+  ): Promise<(input: MultiStepInput) => any> {
     const namespaces: k8s.V1Namespace[] = await listNamespaces();
     const items = namespaces.map((namespace) => {
       return {
         label: `${namespace.metadata?.name}`,
-        description: namespace.metadata?.name,
       };
     });
+    const newNamespace = "New Namespace";
+    const itemsWithNew: vscode.QuickPickItem[] = [
+      { label: newNamespace },
+      ...items,
+    ];
+
     const pick = await input.showQuickPick({
       title,
       step: step,
       totalSteps: totalSteps,
-      placeholder: 'Kubernetes namespace (e.g, myapp)',
-      items: items,
-      activeItem: typeof state.namespace !== 'string' ? state.namespace : undefined,
-      shouldResume: shouldResume
+      placeholder: "Kubernetes namespace",
+      items: itemsWithNew,
+      activeItem: undefined,
+      shouldResume: shouldResume,
     });
+
+    if (pick.label === newNamespace) {
+      return (input: MultiStepInput) => inputNewNamespace(input, state, step);
+    }
+
     state.namespace = pick.label;
     return (input: MultiStepInput) => inputName(input, state, step + 1);
   }
 
+  async function inputNewNamespace(
+    input: MultiStepInput,
+    state: Partial<State>,
+    step: number
+  ) {
+    const ns = await input.showInputBox({
+      title,
+      step: step,
+      totalSteps: totalSteps,
+      validate: async () => undefined,
+      value: "",
+      prompt: "Namespace",
+      shouldResume: shouldResume,
+    });
+    await createNamespace(ns);
+    state.namespace = ns;
```

Comment:
state.namespace = ns.value ?

Reply:
This is for creating a new namespace. state.namespace is the actual namespace we are using. It doesn't make sense to autofill the new space with an actual namespace they might have selected before (that's what state.namespace is). Reasonable to assume that if they want to create a new namespace it's completely new

Reply:
but if they click new namespace, we should populate the state object with the new namespace created, right? not previously selected one, right?

Reply:
Right. `state.namespace = ns;` does that

`value: "",` This part in input.showInputBox is what is populated when the input box is first shown.